consensus/misc/eip4844.go

```diff
@@ -27,13 +30,65 @@ var (
 	dataGaspriceUpdateFraction = big.NewInt(params.BlobTxDataGaspriceUpdateFraction)
 )
 
-// CalcBlobFee calculates the blobfee from the header's excess data gas field.
-func CalcBlobFee(excessDataGas *big.Int) *big.Int {
-	// If this block does not yet have EIP-4844 enabled, return the starting fee
-	if excessDataGas == nil {
-		return big.NewInt(params.BlobTxMinDataGasprice)
+// VerifyEIP4844Header verifies the presence of the excessDataGas field and that
+// if the current block contains no transactions, the excessDataGas is updated
+// accordingly.
+func VerifyEIP4844Header(parent, header *types.Header) error {
+	// Verify the header is not malformed
+	if header.ExcessDataGas == nil {
+		return errors.New("header is missing excessDataGas")
+	}
+	if header.DataGasUsed == nil {
+		return errors.New("header is missing dataGasUsed")
+	}
+	var (
+		parentExcessDataGas uint64
+		parentDataGasUsed   uint64
+	)
+	if parent.ExcessDataGas != nil {
+		parentExcessDataGas = *parent.ExcessDataGas
+		parentDataGasUsed = *parent.DataGasUsed
+	}
+	// Verify that the data gas limit remains within allowed bounds. To check
+	// the **actual** real usage we need to block body (transaction list), but
+	// at least do some sanity checks as early as possible.
+	if parentDataGasUsed > *header.DataGasUsed {
+		if growth := parentDataGasUsed - *header.DataGasUsed; growth > params.BlobTxTargetDataGasPerBlock {
```

Comment:
Intuitively for me these cases are the wrong way around. You're calculating growth if parent > header and shrinkage if parent < header. I guess it depends on how you view it, but I kinda understood shrinkage as how much this header shrank compared to its parent

Reply:
indeed, will fix. my original code didn't have these named and then I started adding names and messed it up .

Reply:
Fixed.